src/utils/parser.js

```diff
@@ -91,40 +111,90 @@ function isFunction(path) {
     t.isObjectMethod(path) || t.isClassMethod(path);
 }
 
-function getFunctions(source: SourceText): Array<SymbolDeclaration> {
-  const ast = getAst(source);
+function getVariableNames(path) {
+  if (t.isObjectProperty(path) && !isFunction(path.node.value)) {
+    return [{
+      name: path.node.key.name,
+      location: path.node.loc
+    }];
+  }
+
+  if (!path.node.declarations) {
+    return path.node.params
+    .map(dec => ({
+      name: dec.name,
+      location: dec.loc
+    }));
+  }
+
+  return path.node.declarations
+    .map(dec => ({
+      name: dec.id.name,
+      location: dec.loc
+    }));
+}
+
+function isVariable(path) {
+  return t.isVariableDeclaration(path) ||
+    (isFunction(path) && path.node.params.length) ||
+    (t.isObjectProperty(path) && !isFunction(path.node.value));
+}
 
-  const functions = [];
+function getSymbols(source: SourceText): SymbolDeclarations {
+  const ast = getAst(source);
+  const symbols = { functions: [], variables: [], classes: [] };
 
   traverse(ast, {
     enter(path) {
+      if (isVariable(path)) {
+        symbols.variables.push(...getVariableNames(path));
+      }
+
       if (isFunction(path)) {
-        functions.push({
+        symbols.functions.push({
           name: getFunctionName(path),
           location: path.node.loc
         });
       }
+
+      if (t.isClassDeclaration(path)) {
+        symbols.classes.push({
+          name: path.node.id.name,
+          location: path.node.loc
+        });
+      }
     }
   });
 
-  return functions;
+  return symbols;
 }
 
-function getFunctionDeclarations(sourceText: SourceText) {
-  if (functionDeclarations.has(sourceText.id)) {
-    return functionDeclarations.get(sourceText.id);
-  }
-
-  const functions = getFunctions(sourceText).map(dec => ({
+function formatDeclarations(
+  declarations: Array<SymbolDeclaration>): Array<FormattedSymbolDeclaration> {
+  return declarations.map(dec => ({
     id: `${dec.name}:${dec.location.start.line}`,
     title: dec.name,
     subtitle: `:${dec.location.start.line}`,
     value: dec.name,
     location: dec.location
   }));
+}
+
+function getSymbolDeclarations(
```

Comment:
Maybe kill this function in place of doing this work within `getSymbols`? I tried keeping it as similar to the previous API.